auth/resource.cc

```diff
@@ -165,6 +214,60 @@ std::ostream &operator<<(std::ostream &os, const service_level_resource_view &v)
     return os;
 }
 
+// Purely for Cassandra compatibility, types in the function signature are
+// decoded from their verbose form (org.apache.cassandra.db.marshal.Int32Type)
+// to the short form (int)
+static sstring decode_signature(std::string_view encoded_signature) {
+    auto name_delim = encoded_signature.find_first_of('[');
+    std::string_view function_name = encoded_signature.substr(0, name_delim);
+    encoded_signature.remove_prefix(name_delim + 1);
+    encoded_signature.remove_suffix(1);
+    std::vector<std::string_view> raw_types;
+    boost::split(raw_types, encoded_signature, boost::is_any_of("^"));
+    std::vector<std::string> decoded_types = boost::copy_range<std::vector<std::string>>(
+        raw_types | boost::adaptors::transformed([] (std::string_view raw_type) {
+            return abstract_type::parse_type(sstring(raw_type))->cql3_type_name();
+        })
+    );
+    return format("{}({})", function_name, boost::algorithm::join(decoded_types, ", "));
```

Comment:
The parentheses here are deliberately different from the brackets above?
Again, please consider what happens if the function name contains a bracket ([) and whether the find_first_of, for example, will break.

Reply:
In my humble opinion, it would be better to just prevent users from using these ridiculous quoted names for functions, and then we'd be done with it, because nobody actually does that anyway. But since we want to be compatible with Cassandra's documentation, I'll ultimately also add code for handling this corner case. I'll do it as a last step though, because I'd rather finish the interesting parts of the implementation, like actually enforcing the permissions, first.

Reply:
Oh and in case of Cassandra, I think it's actually a security risk - I'll double-check, but I'm pretty sure that creating a function named `"some_existing_function[types_of_this_function](int)"` would instead grant you rights to executing `some_existing_function`. Let me try and craft this.

Reply:
Ok, found a security issue in Cassandra, so I'll report it (deleted the original comment because it hinted at the issue)